thirdparty/kyaml/runfn/runfn.go

```diff
@@ -152,37 +132,92 @@ func (r RunFns) getNodesAndFilters() (
 		return nil, nil, outputPkg, err
 	}
 
-	fltrs, err := r.getFilters(buff.Nodes)
+	fltrs, err := r.getFilters()
 	if err != nil {
 		return nil, nil, outputPkg, err
 	}
+
+	// we need to exclude function config file if we don't want to include
+	// meta resources.
+	if !r.IncludeMetaResources {
+		p := kio.Pipeline{
+			Inputs:                []kio.Reader{buff},
+			Filters:               []kio.Filter{kio.FilterFunc(r.filterOutFnConfigFile)},
+			Outputs:               []kio.Writer{buff},
+			ContinueOnEmptyResult: r.ContinueOnEmptyResult,
+		}
+		if err := p.Execute(); err != nil {
+			return nil, nil, outputPkg, err
+		}
+	}
 	return buff, fltrs, outputPkg, nil
 }
 
-func (r RunFns) getFilters(nodes []*yaml.RNode) ([]kio.Filter, error) {
-	var fltrs []kio.Filter
-
-	// fns from annotations on the input resources
-	f, err := r.getFunctionsFromInput(nodes)
-	if err != nil {
-		return nil, err
+// filterOutFnConfigFile is a filter which filters out the function config
+// file
+func (r RunFns) filterOutFnConfigFile(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	var result []*yaml.RNode
+	// fn config path should be absolute
+	var fnConfigPath string
+	if filepath.IsAbs(r.FnConfigPath) {
+		fnConfigPath = r.FnConfigPath
+	} else {
+		// if the FnConfigPath is relative, we should use the
+		// current directory to construct full path.
+		path, err := os.Getwd()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get working directory: %w", err)
+		}
+		fnConfigPath = filepath.Join(path, r.FnConfigPath)
 	}
-	fltrs = append(fltrs, f...)
+	for i := range nodes {
+		node := nodes[i]
+		meta, err := node.GetMeta()
+		if err != nil {
+			return nil, err
+		}
+		path := meta.Annotations[kioutil.PathAnnotation]
+		if path != "" {
+			// file path in the annotation is relative to the package path
+			// we need to use absolute path
+			path = filepath.Join(r.Path, path)
+			if path == fnConfigPath {
+				continue
+			}
+		}
 
-	// fns from directories specified on the struct
-	f, err = r.getFunctionsFromFunctionPaths()
-	if err != nil {
-		return nil, err
+		result = append(result, node)
 	}
-	fltrs = append(fltrs, f...)
+	return result, nil
+}
```

Comment:
One observation: I realized this can be generalized to a `kio.Filter` that ignores a specific `filePath` or `filePaths`.

Reply:
Yes we can do that but that only provides few benefit for us. Actually I think ignoring a list of file paths should be implemented in `kio.LocalPackageReadWriter`. It has capability to ignore files defined in `.krmignore` and only include the files that matched by a list of expressions. It should have a feature to filter out a list of file paths so other package consumers will not need to add a filter manually.